config/v1/types_cluster_version.go

```diff
@@ -408,6 +422,15 @@ type URL string
 // Update represents an administrator update request.
 // +k8s:deepcopy-gen=true
 type Update struct {
+	// architecture is an optional field that indicates the desired
+	// value of the cluster architecture. In this context cluster
+	// architecture means either homogeneous or heterogeneous.
+	// Architecture is ignored if image is specified. The empty
+	// value indicates that the field is not set.
+	//
+	// +optional
+	Architecture ClusterVersionArchitecture `json:"architecture"`
```

Comment:
Why does this need to be specified for every update request?

Reply:
it does not need to be specified, which is why it is `+optional`

Reply:
Cute, but doesn't address the problem.  If I want an architecture for my cluster, why is the architecture not "sticky" across whatever updates I may choose?  it seems more natural to say, "my cluster has this architecture" and separately, "now please update to this version".

Reply:
Ties into my above comment as well. Arch is "sitcky" in that once it's set to `multi` it doesn't need to be set agai since it can only be set to `multi` anyway.

Reply:
> Ties into my above comment as well. Arch is "sitcky" in that once it's set to `multi` it doesn't need to be set agai since it can only be set to `multi` anyway.

Where is this other arch field?  The DesiredUpdate field where I remember this being used is set every time that I want to issue an update and can be cleared to nil as I recall.

Reply:
CVO knows the current architecture by extracting it from the current payload, i.e. if it's "multi" otherwise it sets it using GOARCH as before. Since we couldn't come up with any other consumer of current Architecture we see no need to have it in ClusterVersion status.